src/irsForms/F8959.ts

```diff
@@ -0,0 +1,138 @@
+import { Information } from '../redux/data'
+import { displayNumber, computeField, sumFields } from './util'
+import TaxPayer from '../redux/TaxPayer'
+import Form, { FormTag } from './Form'
+import F4137 from './F4137'
+import F8919 from './F8919'
+import ScheduleSE from './ScheduleSE'
+import { fica } from '../data/federal'
+
+export const needsF8959 = (state: Information): boolean => {
+  const filingStatus = state.taxPayer.filingStatus
+  const totalW2Income = state.w2s.map((w2) => w2.income).reduce((l, r) => l + r, 0)
+  return filingStatus !== undefined &&
+    fica.additionalMedicareTaxThreshold(filingStatus) < totalW2Income
+}
+
+export default class F8959 implements Form {
+  tag: FormTag = 'f8959'
+  state: Information
+  f4137?: F4137
+  f8919?: F8919
+  scheduleSE?: ScheduleSE
+
+  constructor (state: Information, f4137?: F4137, f8919?: F8919, scheduleSE?: ScheduleSE) {
+    this.state = state
+    this.f4137 = f4137
+    this.f8919 = f8919
+    this.scheduleSE = scheduleSE
+  }
+
+  thresholdFromFilingStatus = (): number | undefined => {
+    const filingStatus = this.state.taxPayer.filingStatus
+    return filingStatus !== undefined ? fica.additionalMedicareTaxThreshold(filingStatus) : undefined
+  }
+
+  computeAdditionalMedicareTax = (compensation: number | undefined): number | undefined => {
+    return 0.009 * computeField(compensation)
```

Comment:
Thanks for your contribution! This should be rounded to an integer. The current result has a floating-point error

![image](https://user-images.githubusercontent.com/8848557/125146418-87c10b80-e0da-11eb-8380-abaa6b606d4b.png)

Reply:
Also I would add that .009 parameter deserves a place in the federal.ts file with the rest of our constants that will probably change year to year. 

`irsForms/util` has other functions for displaying numbers, also room for improvement in there, I think using those functions is a bit cumbersome to do math.

Good catch on the rounding issue, otherwise looks good to me.

Reply:
Good catch on the rounding issue! I also moved the tax rate to the data file for consistency, though this one actually doesn't really change year-to-year.